### PR TITLE
netstats: introduce L2 netstats

### PR DESCRIPTION
**Pull request by OlegHahm**
A first step towards addressing https://github.com/RIOT-OS/RIOT/issues/1297.

~~Depends on #4645.~~

Waits for #5176, but is not yet rebased on it.

### CODE REVIEW

**Comment by authmillenon** [comment]
Sorry, I know you already took this out of #4795, but could you maybe make an independent PR, that only introduces `netstat` and then provide the adaptations to the existing `netdev2`, so the introduction PR is not dependent on my PRs (and resulting from that easier and faster to review)?

**Comment by OlegHahm** [comment]
For the record, I'm working on an revised version that's completely independent from other PRs, but includes stats for link and network layer. Stay tuned.

**Comment by OlegHahm** [comment]
Rebased to master, ready for review. To test add `USEMODULE += netstats` to your network application and check with `ifconfig`.

File: sys/include/net/netopt.h [hunk 1]
```diff
@@ -199,6 +199,11 @@ typedef enum {
      */
     NETOPT_CCA_THRESHOLD,
 
+    /**
+     * @brief get statistics about sent and received packets of the layer
+     */
```
Comment (authmillenon) [inline sys/include/net/netopt.h]: Please document the type needed for this option.

File: sys/include/net/netstats.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+extern "C" {
+#endif
+
+/**
+ * @brief       Global statistics struct
+ */
+typedef struct {
+    uint32_t tx_unicast_count;  /**< packets sent via unicast */
+    uint32_t tx_mcast_count;    /**< packets sent via multicast
+                                     (including broadcast */
```
Comment (authmillenon) [inline sys/include/net/netstats.h]: )

**Comment by OlegHahm** [comment]
Sorry, forgot about the discussion in https://github.com/RIOT-OS/RIOT/pull/4801#issuecomment-183434698. Factored it out in #5176.

**Comment by OlegHahm** [comment]
Rebased to current master.

**Comment by OlegHahm** [comment]
Squashed

**Pull request merged**